handler/schedule.go

```diff
@@ -125,3 +137,65 @@ func (s *ScheduleHandler) mergeResourcesWithDefaults(resources corev1.ResourceRe
 	}
 	return resources
 }
+
+func (s *ScheduleHandler) updateSchedule() error {
+	if err := s.Client.Update(s.CTX, s.schedule); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("error updating resource %s/%s: %w", s.schedule.Namespace, s.schedule.Name, err)
+	}
+	return nil
+}
+
+func (s *ScheduleHandler) updateStatus() error {
+	err := s.Client.Status().Update(s.CTX, s.schedule)
+	if err != nil {
+		s.Log.Error(err, "Could not update SyncConfig.", "name", s.schedule)
+		return err
+	}
+	s.Log.Info("Updated SyncConfig status.")
+	return nil
+}
+
+func (s *ScheduleHandler) getOrGenerateSchedule(jobType k8upv1alpha1.JobType, originalSchedule string) string {
+	if s.schedule.Status.GeneratedSchedules == nil {
+		s.schedule.Status.GeneratedSchedules = make(map[k8upv1alpha1.JobType]string)
+	}
+	if schedule, found := s.schedule.Status.GeneratedSchedules[jobType]; found {
+		return schedule
+	}
+	newSchedule := s.randomizeSchedule(s.schedule.Namespace+"/"+s.schedule.Name+"@"+string(jobType), originalSchedule)
+	s.schedule.Status.GeneratedSchedules[jobType] = newSchedule
+	s.requireStatusUpdate = true
+	return newSchedule
+}
+
+// randomizeSchedule randomizes the given originalSchedule with a seed. The originalSchedule has to be one of the supported
+// '@x-random' predefined schedules, otherwise it returns originalSchedule unmodified.
+func (s *ScheduleHandler) randomizeSchedule(seed, originalSchedule string) (schedule string) {
+	hash := sha1.New()
+	hash.Write([]byte(seed))
+	sum := hex.EncodeToString(hash.Sum(nil))
+	sumBase16, _ := new(big.Int).SetString(sum, 16)
+	defer s.Log.V(1).Info("Randomized schedule", "seed", seed, "sum", sum, "from_schedule", originalSchedule, "new_schedule", &schedule)
+
+	minute := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(60))
+	hour := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(24))
+	// A month can have between 27 and 31 days. Cron does not automatically schedule at the end of the month if 31 in February.
+	// To not cause a skip of a scheduled backup that could potentially raise alerts, we cap the day-of-month to 27 so it fits in all months.
+	dayOfMonth := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(27))
+	month := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(12))
+	weekday := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(6))
+	switch originalSchedule {
+	case ScheduleHourlyRandom:
+		return fmt.Sprintf("%d * * * *", minute)
+	case ScheduleDailyRandom:
+		return fmt.Sprintf("%d %d * * *", minute, hour)
+	case ScheduleMonthlyRandom:
+		return fmt.Sprintf("%d %d %d * *", minute, hour, dayOfMonth.Add(dayOfMonth, big.NewInt(1)))
+	case ScheduleAnnuallyRandom:
+	case ScheduleYearlyRandom:
+		return fmt.Sprintf("%d %d %d %d *", minute, hour, dayOfMonth.Add(dayOfMonth, big.NewInt(1)), month.Add(month, big.NewInt(1)))
+	case ScheduleWeeklyRandom:
+		return fmt.Sprintf("%d %d * * %d", minute, hour, weekday)
+	}
+	return originalSchedule
```

Comment:
A switch without a `default` triggers my internal alert-bells. I therefore suggest:

```suggestion
	case ScheduleWeeklyRandom:
		return fmt.Sprintf("%d %d * * %d", minute, hour, weekday)
	default:
		return originalSchedule
	}
```

Reply:
Also, if a check would be implemented that prevents the whole `randomizeSchedule` function to be executed when there is no need for it (i.e. the originalSchedule does not start with an `@`), then the default case should actually log and return an error. Something along the lines of `fmt.Errorf("'%s' is not a recognised schedule", originalSchedule)`.

Reply:
The compiler complained about missing return statement after the switch clause even with a return statement in the default clause. So, with or without the default clause, the effect is the same so I left it out on purpose since it's redundant.

Reply:
I combined your and Simon's feedback. Better?